consensus/hotstuff/vote_aggregator.go

```diff
@@ -2,27 +2,39 @@ package hotstuff
 
 import (
 	"github.com/onflow/flow-go/consensus/hotstuff/model"
-	"github.com/onflow/flow-go/model/flow"
 )
 
-// VoteAggregator aggregates votes and produces quorum certificates.
+// createVote holds the data for signing a block. The constructor of this function should have the block
+// info already, taking block again as argument is to ensure it's called with the same block.
+// For consensus cluster, the createVote function will create a disposable stateful signer object
+// for producing the signature for a vote. After the the signature has been produced, the signer object will
+// be discarded.
+type createVote func(*model.Block) (*model.Vote, error)
+
+// VoteAggregator verifies, aggregates votes, as well as votes for blocks.
```

Comment:
I think we should be very precise here with the wording: 
* Technically, the `Voter` votes for blocks, which is a _different_ component than the `VoteAggregator`. 
* Maybe we could say that the VoteAggregator implements a _factory_ for VoteCollectors

Reply:
Good point.

>Maybe we could say that the VoteAggregator implements a factory for VoteCollectors

I think the comment here is for the user of this interface, instead of the internal modules of this module. For the user of this interface is the compliance engine, which doesn't care about vote collectors or factory.